### PR TITLE
Replace contains with ILike function

### PR DESCRIPTION
**Pull request by elcorbs**
Using `EF.Functions.ILike` results in `ILIKE` being used in the compiled SQL, whereas using `.Contains()` was resulting in the SQL function `STRPOS()` being used in the where statement to check the index of a substring. This is much slower than using `ILIKE`.

### CODE REVIEW

**Review by makeyw — APPROVED** [review]
Well that's a good gotcha to know!

File: AcademyResidentInformationApi/V1/Gateways/AcademyGateway.cs [hunk 1]
```diff
@@ -46,9 +50,9 @@ where string.IsNullOrEmpty(lastname) || person.LastName.ToLower().Replace(" ", "
                 }
                 ).Skip(cursor).Take(limit).ToList().ToDomain();
         }
-        private static string StripString(string str)
+        private static string GetSearchPattern(string str)
         {
-            return str?.ToLower().Replace(" ", "");
+            return $"%{str?.ToLower().Replace(" ", "")}%";
```
Comment (louisebolden) [inline AcademyResidentInformationApi/V1/Gateways/AcademyGateway.cs]: Do we still need to call `.ToLower()` if the result will be used in an `ILIKE` statement?

Comment (elcorbs) [reply AcademyResidentInformationApi/V1/Gateways/AcademyGateway.cs]: Good point, No we don't

**Review by Renny4Real — APPROVED** [review]
👍

**Pull request merged**